python-threatexchange/threatexchange/cli/experimental_fetch.py

```diff
@@ -36,75 +44,118 @@ def init_argparse(cls, ap) -> None:
             type=int,
             help="Fetch updates that occured before this timestamp",
         )
-        ap.add_argument(
-            "--owner",
-            type=int,
-            help="Only fetch updates for indicators that the given app has a descriptor for",
-        )
         ap.add_argument(
             "--threat-types",
             nargs="+",
             help="Only fetch updates for indicators of the given type",
         )
         ap.add_argument(
-            "--additional-tags",
-            nargs="+",
-            help="Only fetch updates for indicators that have a descriptor with each of these tags",
+            "--page-size",
+            type=int,
+            help="The number of updates to fetch per request, defaults to 500",
+            default=500,
         )
 
     def __init__(
         self,
+        continuation: bool,
         start_time: int,
         stop_time: int,
-        owner: int,
         threat_types: t.List[str],
-        additional_tags: t.List[str],
+        page_size: int,
     ) -> None:
+        self.continuation = continuation
         self.start_time = start_time
         self.stop_time = stop_time
-        self.owner = owner
         self.threat_types = threat_types
-        self.additional_tags = additional_tags
-        self.signal_types_by_name = {
-            name: signal() for name, signal in meta.get_signal_types_by_name().items()
-        }
+        self.limit = page_size
         self.last_update_printed = 0
-        self.counts = collections.Counter()
 
     def execute(self, dataset: Dataset) -> None:
-        # TODO: [Potential] Force full fetch if it has been 90 days since the last fetch.
-        self.start_time = (
-            dataset.get_indicator_checkpoint()
-            if self.start_time is None
-            else self.start_time
-        )
-        self.stop_time = int(time.time()) if self.stop_time is None else self.stop_time
-        dataset.load_indicator_cache(self.signal_types_by_name.values())
-        more_to_fetch = True
-        next_page = None
+        request_time = int(time.time())
+        privacy_group = dataset.config.privacy_groups[0]
+        for privacy_group in dataset.config.privacy_groups:
+            self.counts = collections.Counter()
+            self.total_count = 0
+            self.deleted_count = 0
+            self.indicator_store = dataset.get_indicator_store(privacy_group)
+            dataset.load_indicator_cache(self.indicator_store)
+            self.stop_time = request_time if self.stop_time is None else self.stop_time
+            next_page = None
+            if self.continuation:
+                checkpoint = dataset.get_indicator_checkpoint(privacy_group)
+                self.start_time = checkpoint["last_stop_time"]
+                self.stop_time = request_time
+                self.threat_types = checkpoint["threat_types"]
```

Comment:
I don't think threat_type is a concept in the API. This appears to be indicator type? https://developers.facebook.com/docs/threat-exchange/reference/apis/threat-indicator/v9.0

Reply:
Ah I see, unfortunately this is a problem with the threat_updates API and not this python code

Reply:
Updated both to be "types" instead, may take some time to propagate